### ISSUE TITLE
prov/verbs: ibv_req_notify_cq needs to be "primed" and "rearmed"

### ISSUE THREAD

**Issue by dardok** [ofiwg/libfabric#795]
If an application is waiting on the underlying cq wait object from the verbs provider, it will not get the first notification since `fi_ibv_cq_reset` is not called until there's at least one call to `fi_ibv_cq_sread`.  

Also, `fi_ibv_cq_reset` might not be called upon return from `fi_ibv_cq_sread` (i.e. when threshold == 1), and thus the cq event is not drained and notification re-requested.

But, I have another issue that might require restructuring `fi_ibv_cq_sread` (see #796)

**Comment by shefty** [comment]
The latest verbs provider should have fixed this.
